### ISSUE TITLE
SC2016 False positive with `read` into variable-indexed array

### ISSUE THREAD

**Issue by hyperupcall** [koalaman/shellcheck#2693]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash
arr=(a b)
read -r 'arr[1]' <<< 'foo'
```

#### Here's what shellcheck currently says:

```txt
read -r 'arr[$i]' <<< 'foo'
        ^-- [SC2016](https://www.shellcheck.net/wiki/SC2016) (info): Expressions don't expand in single quotes, use double quotes for that.
```

(there are also an 'i' is unused false positive, but that's not what I'm talking about)

#### Here's what I wanted or expected to see:

Nothing

**Comment by thedward** [comment]
Not that you're wrong, but why do you want the single quotes there?

**Comment by hyperupcall** [comment]
> Not that you're wrong, but why do you want the single quotes there?

It feels and looks cleaner and is easier to type.

To be honest, I'm not really sure why `SC2016` is a rule, but in this particular case the intent is clear due to the `read`
